### PR TITLE
Removed aspectj dependency for box parsing.

### PR DESCRIPTION
**Pull request by aldenml**
Fixed write of entries count in AbstractSampleEncryptionBox.

This PR is related to the issue #59 and it's main objective is a discussion. I'm aware that removing the AOP framework is in the scope of development design decisions and future goals should be reviewed.

I didn't perform the ultimate cleanup (of attributes) since it's very intrusive and could slow the review process.

### CODE REVIEW

**Comment by aldenml** [comment]
Hi @sannies , just wanted to know if you think this PR will have an opportunity. Thanks.

**Comment by louisptremblay** [comment]
Hi Aiden, that fork was extremely useful, I couldn't figure out how to make Aspectj work correctly when compiling the project from source into an Adobe AIR native extension. With your change, it worked directly, thanks a lot!!

**Comment by sannies** [comment]
Hi Alden, Louis,

I still owe you a response.  The main disadvantage of the AspectJ-less
solution ist the parsing speed.  With aspectJ details of a box are only
read when you really need them.  Without it many boxes are parsed in full
detail even though they are not required to 'understand'  the file.
I'm still thinking about this problem from time to time but I didn't yet
come to a conclusion on how to remove aspectj but keep the  on demand
parsing.

Best Regards,
Sebastian

louisptremblay <notifications@github.com> schrieb am Mi., 27. Mai 2015
04:53:

**Comment by aldenml** [comment]
@sannies I totally understand you. If I figure out a way to lazy load the boxes, do you think this PR will fly? Or you still are worried about the mdat boxes?

Regards

**Comment by aldenml** [comment]
@louisptremblay Good to know it works for you.

**Comment by gubatron** [comment]
+1 to this PR.
In the meantime, I believe I'll be forking this and including this PR.

>"Without it many boxes are parsed in full detail even though they are not required to 'understand'  the file."

"many"...
I think it's worth it running some stats on a number of videos to see what "many" in real life actually is.
Perhaps with some solid data we can convince you and then we won't have to fork. This aspectj nonsense is overkill.

File: isoparser/src/main/java/com/googlecode/mp4parser/RequiresParseDetailAspect.java [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-@Aspect
-/**
- * An aspect to trigger the actual parsing of boxes' content when it is actually needed. This aspect
- * calls {@link com.googlecode.mp4parser.AbstractBox#parseDetails()} before actually executing
- * the method.
- */
-public class RequiresParseDetailAspect {
-
-
-    @Before("this(com.googlecode.mp4parser.AbstractBox) && ((execution(public * * (..)) && !( " +
```
Comment (gubatron) [inline isoparser/src/main/java/com/googlecode/mp4parser/RequiresParseDetailAspect.java]: having this seems like more of an overhead than not having it...

**Comment by frostwire** [comment]
In the meantime, for those interested in using mp4parser on android and removing the overhead of box checking with aspectj, please use this [mp4parser fork](https://github.com/frostwire/mp4parser/)

**Pull request closed without merging**